ResearchKit/ActiveTasks/ORKWaitStepViewController.m

```diff
@@ -0,0 +1,218 @@
+    
+    if (_progressView) {
+        NSDictionary *screenMetric = @{@"progressWidth": [NSNumber numberWithFloat:([UIScreen mainScreen].bounds.size.width - 40.0)]};
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(20.0)-[_progressView(progressWidth)]-(20.0)-|" options:NSLayoutFormatAlignAllBaseline metrics:screenMetric views:NSDictionaryOfVariableBindings(_progressView)]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_progressView]|" options:NSLayoutFormatAlignAllCenterX metrics:nil views:NSDictionaryOfVariableBindings(_progressView, _textLabel)]];
+    } else if (_activityIndicatorView) {
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_activityIndicatorView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0.0]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(>=0)-[_activityIndicatorView]-(>=0)-|" options:NSLayoutFormatAlignAllBaseline metrics:nil views:NSDictionaryOfVariableBindings(_activityIndicatorView)]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_activityIndicatorView]|" options:NSLayoutFormatAlignAllCenterX metrics:nil views:NSDictionaryOfVariableBindings(_activityIndicatorView, _textLabel)]];
+    }
+}
+
+#pragma mark Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    if (_progressView) {
+        return ORKAccessibilityStringForVariables(_textLabel.accessibilityLabel, _progressView.accessibilityLabel);
+    } else if (_activityIndicatorView) {
+        return ORKAccessibilityStringForVariables(_textLabel.accessibilityLabel, _activityIndicatorView.accessibilityLabel);
+    }
+    return nil;
+}
+
+- (UIAccessibilityTraits)accessibilityTraits {
+    if (_progressView) {
+        return [super accessibilityTraits] | UIAccessibilityTraitUpdatesFrequently;
+    } else {
+        return [super accessibilityTraits];
+    }
+}
+
+@end
+
+@interface ORKWaitStepViewController ()
+
+@property (nonatomic, strong) ORKWaitView *waitView;
+
+@end
+
+@implementation ORKWaitStepViewController {
+    ORKProgressIndicatorMask _indicatorMask;
+}
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    if (self) {
+        self.suspendIfInactive = NO;
+    }
+    return self;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.learnMoreButtonItem = nil;
+    self.activeStepView.headerView.captionLabel.text = nil;
+
+    _waitView = [[ORKWaitView alloc] initWithIndicatorMask:((ORKWaitStep *)self.step).indicatorMask heading:self.step.title];
+    _waitView.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
Maybe we shouldn't alloc init in here. We can just do it inside `stepDidChange`.